File: idea-plugin/src/github/jk1/smtpidea/ui/DetailedMailViewDialog.java
package github.jk1.smtpidea.ui;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.ui.components.JBScrollPane;
import com.intellij.ui.components.JBTabbedPane;
import github.jk1.smtpidea.server.MailSessionInfo;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 *
 */
public class DetailedMailViewDialog extends DialogWrapper {

    private MailSessionInfo info;
    private JBTabbedPane tabbedPane;

    protected DetailedMailViewDialog(@Nullable Project project, MailSessionInfo info) {
        super(project);
        this.setTitle("Incoming Mail View");
        this.info = info;
        init();
    }

    @NotNull
    @Override
    protected JComponent createCenterPanel() {
        tabbedPane = new JBTabbedPane();
        this.addTab("Raw", info.getRawMessage());
        this.addTab("Text", info.getFormattedMessage());
        this.addTab("Html", info.getFormattedMessage());
        return tabbedPane;
    }

    private void addTab(String title, String data) {
        JPanel panel = new JPanel();
        JTextPane pane = new JTextPane();
        pane.setText(data);
        panel.add(new JBScrollPane(pane));
        tabbedPane.addTab(title, panel);
    }
}

File: src/main/kotlin/github/jk1/smtpidea/ui/DetailedMailViewDialog.kt
package github.jk1.smtpidea.ui

import com.intellij.openapi.ui.DialogWrapper
import com.intellij.openapi.project.Project
import com.intellij.ui.components.JBTabbedPane
import javax.swing.JComponent
import javax.swing.JPanel
import javax.swing.JTextPane
import com.intellij.ui.components.JBScrollPane
import github.jk1.smtpidea.server.smtp.MailSession

/**
 * Dialog showing detailed message contents: headers, body and so on
 */
public class DetailedMailViewDialog(project: Project?, val info: MailSession) : DialogWrapper(project){

    private val tabbedPane = JBTabbedPane();

    {
        setTitle("Incoming Mail View")
        init()
    }

    override fun createCenterPanel(): JComponent {
        addTab("Raw", info.getRawMessage())
        addTab("Text", info.getFormattedMessage())
        return tabbedPane
    }

    private fun addTab(title: String, data: String) {
        val panel = JPanel()
        val pane = JTextPane()
        pane.setText(data)
        panel.add(JBScrollPane(pane))
        tabbedPane.addTab(title, panel)
    }
}
